### PR TITLE
Tokenomics suite

### PR DESCRIPTION
**Pull request by Jayashsatolia403**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -94,34 +90,30 @@ jobs:
...
         if: github.event_name == 'workflow_dispatch' && github.event.inputs.existing_network != ''
         run: |
           echo "NETWORK_URL=$(echo ${{ github.event.inputs.existing_network }})" >> $GITHUB_ENV
           echo "ZBOX_BRANCH=$(echo ${{github.event.inputs.zbox_cli_branch}})" >> $GITHUB_ENV
           echo "ZWALLET_BRANCH=$(echo ${{github.event.inputs.zwallet_cli_branch}})" >> $GITHUB_ENV
+          
+          echo "TEST_FILE_FILTER=false" >> $GITHUB_ENV
 
-          echo "TEST_FILE_FILTER=$(echo $(([ -z '${{github.event.inputs.test_file_filter}}' ] && echo '') || echo '${{github.event.inputs.test_file_filter}}'))" >> $GITHUB_ENV
+          echo "TOKENOMICS_ALLOCATION_TESTS=$(echo $(([ -z 'TestAllocation' ] && echo '') || echo 'TestAllocation'))" >> $GITHUB_ENV
```
Comment (Kishan-Dhakan) [inline .github/workflows/ci.yml:110]: Please migrate this workflow to a different file and revert this ci.yaml, idea being that we want a tokenomics suite to be trigerred manually or through crons only, not on every commit.

**Review by Kishan-Dhakan — CHANGES_REQUESTED** [review]
Minor refactor needed, awesome PR

**Comment by stewartie4** [comment]
Reviewing this shortly

File: .github/workflows/tokenomics_ci.yml [hunk 2]
```diff
@@ -0,0 +1,473 @@
+name: "0Chain System Tests"
```
Comment (Kishan-Dhakan) [inline .github/workflows/tokenomics_ci.yml:1]: Please rename this to `Zus Tokenomics System Tests`

**Review by Kishan-Dhakan — CHANGES_REQUESTED** [review]
Partial review

File: test-output.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
+{"Time":"2023-05-13T00:54:38.445272933+05:30","Action":"start","Package":"github.com/0chain/system_test/internal/api/model"}
+{"Time":"2023-05-13T00:54:38.445376049+05:30","Action":"output","Package":"github.com/0chain/system_test/internal/api/model","Output":"?   \tgithub.com/0chain/system_test/internal/api/model\t[no test files]\n"}
```
Comment (stewartie4) [inline test-output.json:2]: this can be removed

File: tests/tokenomics_tests/test-output.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{"Time":"2023-05-13T00:55:50.355010076+05:30","Action":"start","Package":"github.com/0chain/system_test/tests/tokenomics_tests"}
+{"Time":"2023-05-13T00:55:50.355133007+05:30","Action":"start","Package":"github.com/0chain/system_test/tests/tokenomics_tests/utils"}
```
Comment (stewartie4) [inline tests/tokenomics_tests/test-output.json:2]: this can be removed

File: .github/workflows/tokenomics_ci.yml [hunk 5]
```diff
@@ -0,0 +1,473 @@
+name: "Zus Tokenomics System Tests"
```
Comment (stewartie4) [inline .github/workflows/tokenomics_ci.yml:1]: should this run on cron on a nightly basis if not against the CI

File: tests/tokenomics_tests/config/wallets/UTC--2022-11-13T22-51-36.015115000Z--d8c9156e782c68ee671c09b6b92de76c97948432 [hunk 6]
```diff
@@ -0,0 +1 @@
+{"address":"d8c9156e782c68ee671c09b6b92de76c97948432","crypto":{"cipher":"aes-128-ctr","ciphertext":"e4a7f436e5aaf5e70c4d599710d73e87555ca52f8e980609f318cb1b139a5731","cipherparams":{"iv":"e9ab45b3c316f0f1af6279747682787f"},"kdf":"scrypt","kdfparams":{"dklen":32,"n":262144,"p":1,"r":8,"salt":"6a06e705fd938963f61971296034a725051ccc80009053b8840229797a3583d7"},"mac":"12b8e284bc94e18955b534c8ce0883f0bce2d86358322aee3e60fcb288541159"},"id":"3aa1668c-8285-4770-86db-71dd8bcb8c20","version":3}
```
Comment (stewartie4) [inline tests/tokenomics_tests/config/wallets/UTC--2022-11-13T22-51-36.015115000Z--d8c9156e782c68ee671c09b6b92de76c97948432:1]: can these be renamed to something more meaningful at a glance?

File: tests/tokenomics_tests/SUBSET_TEST_RUNNER_COMMAND.sh [hunk 7]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+set -o pipefail
+[ -z "$1" ] && TESTS_TO_RUN="-run ^\QTestAllocation\E$ ./..." || TESTS_TO_RUN=$1
```
Comment (stewartie4) [inline tests/tokenomics_tests/SUBSET_TEST_RUNNER_COMMAND.sh:3]: this should live in the actions repo

**Review by stewartie4 — COMMENTED** [review]
First scan looks great, if you need anything to help get this across the line let me know 
Amazing coverage

File: tests/tokenomics_tests/config/wallets/UTC--2022-11-13T22-51-36.015115000Z--d8c9156e782c68ee671c09b6b92de76c97948432 (hunk 6 above)
Comment (Jayashsatolia403) [reply tests/tokenomics_tests/config/wallets/UTC--2022-11-13T22-51-36.015115000Z--d8c9156e782c68ee671c09b6b92de76c97948432:1]: These are from cli tests only, I didn't add them.

File: tests/tokenomics_tests/SUBSET_TEST_RUNNER_COMMAND.sh (hunk 7 above)
Comment (Jayashsatolia403) [reply tests/tokenomics_tests/SUBSET_TEST_RUNNER_COMMAND.sh:3]: We had a good discussion to keep it here only. @Kishan-Dhakan can clarify.

**Pull request closed without merging**